### ISSUE TITLE
[Bug]: Outline Manager cannot create new valid keys

### ISSUE THREAD

**Issue by pavelgivoy** [Jigsaw-Code/outline-apps#2292]
### Application

Outline Manager

### Describe the bug

The Outline Manager cannot create new valid keys to use VPN server. The server is running and alive.

### Steps to reproduce

1. Open the Outline Manager
2. Choose my server
3. Create a new key and get a response that the key cannot be created
![изображение_2024-11-23_113125570](https://github.com/user-attachments/assets/50f8dc13-6a76-4cc5-97ed-e69307c3ffc8)
4. Close the Outline Manager
5. Open the Outline Manager again, and I see that the key is created
![изображение_2024-11-23_113219630](https://github.com/user-attachments/assets/e05228ec-9fd9-44fe-8a16-b373875c0f4e)
6. Open the Outline client
7. Insert the created key here and get the response that the key is invalid
![изображение_2024-11-23_113319612](https://github.com/user-attachments/assets/3f5c06da-aea7-4eb6-af78-6e5c5077d9db)

### What did you expect to happen?

I expect that the valid key is created and can be used for new connections

### What actually happened?

All created keys are invalid

### Outline Version

1.15.2 (Manager); 1.12.0 (Client Windows); 1.13.0 and 1.14.0 (Client iOS)

### What operation system are you using?

iOS, Windows

### Operating System Version

Windows 10 22H2 (Manager and Client)
iOS 18.0.1 (Client)
iOS 17.7.6 (Client)

### Screenshots and Videos

All screenshots are given above at reproducing steps unit

P.S. I already have three keys for the current server, that are successfully working. This problem is actual for the newer keys.

**Comment by sbruens** [comment]
It's hard to tell whether this is a Manager application issue or a server API issue. Are you able to use the API directly to try and create a key and see the response? Here are instructions on how to use the API: https://github.com/Jigsaw-Code/outline-server/tree/master/src/shadowbox#access-keys-management-api

**Comment by pavelgivoy** [comment]
I created the key directly via API. The created key is shown when I use API command to list all keys. Get command also shows that key. When I put it into Outline Client, it says that the key is invalid

**Comment by sbruens** [comment]
Can you validate the format of the key you created? For example, I'd expect it to look something like this: `ss://Y2hhY2hhMjAtaWV0Zi1wb2x5MTMwNTo1UkVmeFRqbHR6Mkw@1.2.3.4:17178`.  Make sure the IP (the `1.2.3.4` example in the URL above) is set.

You can also inspect the docker logs from around the time you created the key, or create a new key and check the logs to see if the same error happens. That will help us understand what error is causing this. You can run `docker logs shadowbox` from inside the server to get the logs.

My guess of what's happening is that there is a problem happening [here](https://github.com/Jigsaw-Code/outline-apps/blob/23afedf81b3873bac08f60ceb4ed36404f4c44b4/server_manager/www/app.ts#L1114C1-L1125C6), where the key is getting created but not cleaned up in the error case. We could show a better error message in the Manager. Although I'm surprised the API doesn't error on creating a new key either. Seeing the docker logs will help pinpoint the problem.

**Comment by pavelgivoy** [comment]
I created the key via API. The operation is completed successfully. The key is shown via list and get API commands. This key is also shown in the Outline Manager app. However, when I insert the key into Outline Client, I get answer that the key is invalid. The server is working. Moreover, I have three keys for the server, which are working stable.

**Comment by pavelgivoy** [comment]
The key format satisfies your example. The IP address is set.

According to logs, the last entry in them is:
`ts=2024-11-14T04:00:53.021Z caller=scrape.go:1277 level=error component="scrape manager" scrape_pool=outline-server-main target=http://127.0.0.1:9091/metrics msg="Scrape commit failed" err="write to WAL: log samples: write /opt/outline/persisted-state/prometheus/data/wal/00000009: no space left on device"`

I don't have newer entries, and I cannot know what the server does with created keys. In my suggestion, the server doesn't create them because of lack of disk space. Bot it's better for you to make your own analysis of my issue.

**Comment by sbruens** [comment]
Yes, it seems likely the lack of disk space is causing issues. It's probably unable to update the `outline-ss-server` config with the new key because that requires writing to disk. Resolving the disk space issue first will help you debug further, but that may resolve it completely.

**Comment by sbruens** [comment]
Hi there,

We're closing this issue due to inactivity. We haven't heard back from you in a while, and aim to keep our issue tracker organized.

If you're still experiencing this problem or have any additional information to share, please feel free to reopen this issue or create a new one.

Thank you for your understanding!
